includes/data-port/class-sensei-data-port-manager.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * File containing the Sensei_Data_Port_Manager class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * This class is responsible creating, deleting and keeping track of data port jobs.
+ */
+class Sensei_Data_Port_Manager implements JsonSerializable {
+	const OPTION_NAME = 'sensei-tools-jobs';
+
+	/**
+	 * An array of all in progress data port jobs. It has the following format:
+	 * {
+	 *
+	 *     @type string $id        Unique id for this job.
+	 *     @type int    $user_id   The user which initiatied this job.
+	 *     @type int    $time      When the job started.
+	 *     @type string $handler   The class which handles this job.
+	 * }
+	 *
+	 * @var array
+	 */
+	private $data_port_jobs;
+
+	/**
+	 * Tracks if the data port jobs have been updated.
+	 *
+	 * @var boolean
+	 */
+	private $has_changed;
+
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Data_Port_Manager constructor.
+	 */
+	private function __construct() {
+		$json_string       = get_option( self::OPTION_NAME );
+		$this->has_changed = false;
+
+		if ( $json_string ) {
+			$this->data_port_jobs = json_decode( $json_string, true );
+		} else {
+			$this->data_port_jobs = [];
+		}
+	}
+
+	/**
+	 * Initializes the data port manager.
+	 */
+	public function init() {
+		foreach ( $this->data_port_jobs as $job ) {
+			add_action( Sensei_Data_Port_Job::SCHEDULED_ACTION_NAME, [ $this, 'run_data_port_job' ] );
+		}
+
+		add_action( 'shutdown', [ $this, 'persist' ] );
+	}
+
+	/**
+	 * Runs a data port job.
+	 *
+	 * @param array $args  The arguments of the background job. Only the job_id is included.
+	 */
+	public function run_data_port_job( $args ) {
+
+		if ( empty( $args['job_id'] ) ) {
+			return;
+		}
+
+		$job = $this->get_job( $args['job_id'] );
+
+		if ( null !== $job ) {
+			Sensei_Scheduler::instance()->run( $job );
+		}
+	}
+
+	/**
+	 * Starts a data import job.
+	 *
+	 * @param int $user_id  The user which started the job.
+	 */
+	public function start_import_job( $user_id ) {
```

Comment:
We're going to want to split starting and initializing.

We'll need to start tracking things like files and options in a job before we schedule it.

Reply:
Yep. I wrote this when I didn't had the extra create action in mind. Added a TODO for now, let's do this when we implement the creation endpoint.